<!-- 
  ⭐ Modified version of `double.launch` for the `aruco_ros` package.
  This file is modified to fit the needs of vS-Graphs. ⭐
-->
 
 <launch>
    <!-- General Parameters -->
    <arg name="side"        default="left"/>
    <!-- Marker size in meters -->
    <arg name="markerSize"  default="0.114"/>
    <!-- If empty, pose will be published w.r.t. camera frame -->
    <arg name="ref_frame"   default=""/>
    <!-- Image Topic -->
    <arg name="imageRaw"    default="/camera/color/image_raw" />
    <!-- Camera Parameters Topic -->
    <arg name="cameraInfo"  default="/camera/color/camera_info" />

    <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
        <!-- Remap -->
        <remap from="/image"                to="$(arg imageRaw)" />
        <remap from="/camera_info"          to="$(arg cameraInfo)" />
        <!-- Parameters -->
        <param name="image_is_rectified"    value="True"/>
        <param name="reference_frame"       value="$(arg ref_frame)"/>
        <param name="marker_size"           value="$(arg markerSize)"/>
        <param name="camera_frame"          value="$(arg side)_hand_camera"/>
    </node>

</launch>